packages/editor/src/components/provider/index.js

```diff
@@ -161,31 +164,57 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 		BlockEditorProviderComponent = ExperimentalBlockEditorProvider,
 		__unstableTemplate: template,
 	} ) => {
-		const { editorSettings, selection, isReady, mode } = useSelect(
-			( select ) => {
+		const { editorSettings, selection, isReady, mode, postTypes } =
+			useSelect( ( select ) => {
 				const {
 					getEditorSettings,
 					getEditorSelection,
 					getRenderingMode,
 					__unstableIsEditorReady,
 				} = select( editorStore );
+				const { getPostTypes } = select( coreStore );
+
 				return {
 					editorSettings: getEditorSettings(),
 					isReady: __unstableIsEditorReady(),
 					mode: getRenderingMode(),
 					selection: getEditorSelection(),
+					postTypes:
+						getPostTypes( { per_page: -1 } )?.map(
+							( entity ) => entity.slug
+						) || [],
```

Comment:
You should avoid `map/filter` operations for returned values from `mapSelect`. They will always return new reference and force component to re-render even if derived value is the same.

P.S. You should see warning in console for similar cases in dev mode. See #53666.

Reply:
Thanks for the info! I wasn't aware of that 🙂 I have just changed it in [this commit](https://github.com/WordPress/gutenberg/pull/65062/commits/ac443f526aac7bc3f2f2cf949e0582b0cd186f34).

Not related to this pull request, but do you think it could be an issue also [here](https://github.com/WordPress/gutenberg/blob/ac443f526aac7bc3f2f2cf949e0582b0cd186f34/packages/block-library/src/query/utils.js#L104-L107)? I'm happy to create another PR to change that.

Reply:
Thank you, @SantosGuillamot!

> Not related to this pull request, but do you think it could be an issue also [here](https://github.com/WordPress/gutenberg/blob/ac443f526aac7bc3f2f2cf949e0582b0cd186f34/packages/block-library/src/query/utils.js#L104-L107)? I'm happy to create another PR to change that.

That one is okay. It will pass a shallow equality check since the `filteredPostTypes` values will have the same reference. It's somewhat undocumented and a last-resort hack.

It usually depends on the case; check for a warning in the DevTools if you're unsure.

```js
// ❌ will fail shallow equality since the `postType` property has a new reference on each call.
return {
   postTypes: types.filter( () => /*--/* );
}

// ✅ will pass shallow equality since returned object properties have the same object reference
return types.filter( () => /*--/* );
```